stock_location_lockdown/models/stock_quant.py

```diff
@@ -3,24 +3,19 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from odoo import api, models, _
-from odoo.exceptions import UserError
+from odoo.exceptions import ValidationError
 
 
 class StockQuant(models.Model):
     _inherit = 'stock.quant'
 
-    @api.model
-    def quants_move(
-            self, quants, move, location_to, location_from=False, lot_id=False,
-            owner_id=False, src_package_id=False, dest_package_id=False,
-            entire_pack=False):
-        if (
-                location_to.usage == 'internal' and
-                location_to.block_stock_entrance):
-            raise UserError(_(
-                "The location '%s' is not configured to receive stock.")
-                % location_to.display_name)
-        return super(StockQuant, self).quants_move(
-            quants, move, location_to, location_from=location_from,
-            lot_id=lot_id, owner_id=owner_id, src_package_id=src_package_id,
-            dest_package_id=dest_package_id, entire_pack=entire_pack)
+    @api.constrains('location_id')
+    def _check_location_blocked(self):
+        for record in self:
+            if record.location_id.block_stock_entrance:
```

Comment:
@sebastienbeau You removed the check on 'usage' field. Why ?

Reply:
@rousseldenis 
I suggested it in my review.
I think it is useless, the boolean `block_stock_entrance` is enought to know the location should not contain any quant.
When I made this module in v8, the case I wanted to resolve was really linked to internal location, and so I guess I put this condition, but actually, it shoud also concern view locations, and why not others...